pkg/app/api/grpcapi/piped_api.go

```diff
@@ -94,8 +94,18 @@ func (a *PipedAPI) Ping(ctx context.Context, req *pipedservice.PingRequest) (*pi
 // ReportStat is periodically sent to report its realtime status/stats to control-plane.
 // The received stats will be pushed to the metrics collector.
 func (a *PipedAPI) ReportStat(ctx context.Context, req *pipedservice.ReportStatRequest) (*pipedservice.ReportStatResponse, error) {
+	_, pipedID, _, err := rpcauth.ExtractPipedToken(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
How about adding a simple guard here to ensure that the sent data is not exceeding the maximum size we can accept?
Just to protect the stat cache.

Reply:
Sure 👍  I wonder which limit we should set it to 🤔 
Suppose we have 100 pipeds commit its stats, then it will be 51.2MB for each? 🤔 

> A String value can be at max 512 Megabytes in length.

ref: https://redis.io/topics/data-types

Reply:
I think 1MB is enough.
But I just remembered that gRPC is also having its validation on the message size.
Its default value is 4MB. So it is good as well.
So I think we don't have to add any additional code, just leave to use the default value.

Reply:
Gotcha 👌 
Another thing I have concerned about is the number of "field" in this hash. As mentioned in the [docs](https://redis.io/topics/data-types)
> A hash with a few fields (where few means up to one hundred or so) is stored in a way that takes very little space, so you can store millions of objects in a small Redis instance.

We may have a limit of up to hundreds of piped which could be registered to controlplane 👀  Will open an issue for that later since up to this time, we don't have any limitation around how may piped can be controlled by controlplane.

Reply:
You are right. We may split the Piped list into smaller groups like grouping by projectID.
Nice to have an issue about the field number.